lib/node_modules/@stdlib/_tools/changelog/parse-commits/lib/conventional_changelog.js

```diff
@@ -0,0 +1,276 @@
+var RE_DIGITS = /^[0-9]+$/;
+var RE_CLOSES = /(?:closes|fixes|resolves)/i;
+var RE_CO_AUTHORED_BY = /co-authored-by/i;
+
+
+// MAIN //
+
+/**
+* Converts a commit message AST to the conventional changelog format.
+*
+* @private
+* @param {Object} ast - unist commit message AST
+* @returns {Object} conventional changelog format
+*/
+function toConventionalChangelog( ast ) {
+	var out = {
+		'type': '',
+		'scope': null,
+		'header': '',
+		'subject': '',
+		'body': '',
+		'notes': [],
+		'references': [],
+		'mentions': [],
+		'footer': null
+	};
+
+	visit( ast, 'body', onBody );
+	visit( ast, 'summary', onSummary );
+	visitWithAncestors( ast, [ 'breaking-change' ], processBreakingChanges );
+	visit( ast, 'footer', processFooter );
+
+	return out;
+
+	/**
+	* Callback invoked upon visiting a breaking change node.
+	*
+	* @private
+	* @param {Object} _ - AST node
+	* @param {ObjectArray} ancestors - node ancestors
+	*/
+	function processBreakingChanges( _, ancestors ) {
+		var startCollecting = false;
+		var breaking = {
+			'title': 'BREAKING CHANGE',
+			'text': ''
+		};
+		var parent = ancestors.pop();
+		switch ( parent.type ) {
+		case 'summary':
+			breaking.text = out.subject;
+			break;
+		case 'body':
+			breaking.text = '';
+			visit( parent, [ 'text', 'breaking-change' ], visitBodyNode );
+			break;
+		case 'token':
+			parent = ancestors.pop();
+			visit( parent, 'text', visitTextNode );
+			break;
+		default:
+			break;
+		}
+		if ( breaking.text !== '' ) {
+			out.notes.push( breaking );
+		}
+
+		/**
+		* Callback invoked upon visiting a body node.
+		*
+		* @private
+		* @param {Object} node - AST node
+		*/
+		function visitBodyNode( node ) {
+			if ( startCollecting && node.type === 'text' ) {
+				if ( breaking.text !== '' ) {
+					breaking.text += '\n';
+				}
+				breaking.text += node.value;
+			} else if ( node.type === 'breaking-change' ) {
+				startCollecting = true;
+			}
+		}
+
+		/**
+		* Callback invoked upon visiting a text node.
+		*
+		* @private
+		* @param {Object} node - AST node
+		*/
+		function visitTextNode( node ) {
+			breaking.text = node.value;
+		}
+	}
+
+	/**
+	* Callback invoked upon visiting the body node.
+	*
+	* @private
+	* @param {Object} body - AST node
+	*/
+	function onBody( body ) {
+		if ( body ) {
+			visit( body, 'text', extractBody );
+		}
+
+		/**
+		* Callback invoked upon visiting a text node.
+		*
+		* @private
+		* @param {Object} node - AST node
+		*/
+		function extractBody( node ) {
+			if ( out.body !== '' ) {
+				out.body += '\n';
+			}
+			out.body += node.value;
+		}
+	}
+
+	/**
+	* Callback invoked upon visiting the summary node.
+	*
+	* @private
+	* @param {Object} summary - AST node
+	*/
+	function onSummary( summary ) {
+		visit( summary, visitSummary );
+
+		/**
+		* Processes a summary node and extracts relevant information.
+		*
+		* @private
+		* @param {Object} node - AST node
+		*/
+		function visitSummary( node ) {
```

Comment:
Same comment. This function should be moved on-level up. Otherwise, for every `onSummary` call, you are creating a new function, which increases memory usage and creates additional GC pressure.

Reply:
Done.